### PR TITLE
Delete JIT_PATH host key/value pair

### PR DESCRIPTION
**Pull request by jkotas**
This was originally introduced to deal with the JIT being in a separate NuGet package. No longer the case for a years.

### CODE REVIEW

**Review by vitek-karas — APPROVED** [review]
I don't have a problem with this, but just to note:
* This is technically a breaking change
* I guess we don't want to support custom JIT in this way, right? (as in plugin a different JIT without copying over the existing one).

**Review by swaroop-sridhar — APPROVED** [review]

**Comment by jkotas** [comment]
> This is technically a breaking change

Do you have anything specific in mind? I believe that only behavior change that somebody can see is that `AppContext.GetData("JIT_PATH")` is going be a null string.

> I guess we don't want to support custom JIT in this way, right?

Right. There are many parts missing to make this work. I do not think anybody would be able to use it in any reasonable way in its current form anyway.

**Comment by vitek-karas** [comment]
> Do you have anything specific in mind? I believe that only behavior change that somebody can see is that AppContext.GetData("JIT_PATH") is going be a null string.

Yes, that part. I've seen people refer to `JIT_PATH` as possible way to locate the runtime. Not that I want to support those use cases... just saying.

**Comment by jkotas** [comment]
> JIT_PATH as possible way to locate the runtime

.FWIW, I can't find any such usages on github.

File: src/installer/corehost/cli/hostpolicy/deps_resolver.cpp [hunk 1]
```diff
@@ -892,7 +886,6 @@ bool deps_resolver_t::resolve_probe_paths(probe_paths_t* probe_paths, std::unord
 
     // If we found coreclr and the jit during native path probe, set the paths now.
     probe_paths->coreclr = m_coreclr_path;
-    probe_paths->clrjit = m_clrjit_path;
```
Comment (elinor-fung) [inline src/installer/corehost/cli/hostpolicy/deps_resolver.cpp:895]: I think `probe_paths_t::clrjit` can go away too.

**Review by elinor-fung — APPROVED** [review]

**Pull request merged**